storage_test.go

```diff
@@ -0,0 +1,11 @@
+package tprstorage
+
+import (
+	"testing"
+
+	"github.com/giantswarm/microstorage"
+)
+
+func TestInterface(t *testing.T) {
```

Comment:
I would like to have more tests here. We have to separate between unit and integration tests. I have no idea how to sort out the integration tests since we can dream this as big as we want. Lets maybe have a dedicated brainstorming session on this. Maybe the host cluster CI story brings also something with it that helps us here. Towards unit tests, it would be good to separate the plain algorithms from external dependencies. For instance `List` does both in the same function. If the algorithm part is separated we could test the critical stuff with checking on slashes (`/`) and creating test cases to get this right without bothering about mocking the k8s client. 

Note that I did some poor mans integration test with Docker back then. See the README and tests in https://github.com/giantswarm/microkit/tree/master/storage/etcd.